core/corehttp/gateway_handler.go

```diff
@@ -63,8 +65,23 @@ func (i *gatewayHandler) loadTemplate() error {
 	return nil
 }
 
-func (i *gatewayHandler) ResolvePath(path string) (*dag.Node, error) {
-	return i.node.Resolver.ResolvePath(path)
+func (i *gatewayHandler) ResolvePath(p string) (*dag.Node, string, error) {
+	if strings.HasPrefix(p, "/ipns/") {
+		var err error
+		p, err = i.node.Namesys.Resolve(path.Clean(p[6:]))
```

Comment:
btw, one thing is it doesnt have to be:

```
<ipns-hash> -> <ipfs-hash>
```

It could be

```
<ipns-hash> -> <ipfs-path>
```

So:

```
# could have mapping:
/ipns/Qmabcd -> /ipfs/Qmjklkj...../a/b/c/d/e

# so can resolve:
/ipns/Qmabcd/foo/bar -> /ipfs/Qmjklkj...../a/b/c/d/e/foo/bar
```